mojitoinc/mojito-mixers
# MOJITO-1052: Refactor to reduce redirects and fix bundling issues.

**Minor changes:**

- Add secondary button href (`goToHref`) to `CheckoutModalFooter`.
- Add missing close link href (`closeHref`) and submit button href (`submitHref`) ( to `CheckoutModalFooter` (already used in `ErrorView`).
- Add `paymentId` param validation and redirects in `SuccessOverlay ` (implemented in its `getServerSideProps` and `CheckoutOverlay` itself, in case some projects don't use SSR).
- Add URL validation and redirects in `ErrorOverlay`.


**Refactor:**

- Use `CheckoutOverlay` in `/payments/success` page and persist `orgID`, `checkoutItems`... from cookie to be able to resume the flow and show the confirmation view there, instead of redirecting to the product page first.
- Remove `CheckoutProvider`.
- Persist cookie with `paymentID` in the key, but doesn't use it to load the data again until we figure out what's going on with the 3DS integration.
- Move success/error/failure page pathnames to constants.


**Testing:**

- Make sure ErrorOverlay passes error message as URL param.
- Make sure 3DS error flow is resumed properly (show error loader variant, display previous data, timer continues, keep item and units...).


**TODO:**

- Show an error when going back to the NFT/item page before completing the 3DS flow. This has been partially fixed (now you don't see the confirmation screen there). I've created a new task to show an actual error too: https://mojitoinc.atlassian.net/browse/MOJITO-1086
